optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/handlers/planner/PlannerDomainHandler.java

```diff
@@ -78,193 +79,180 @@ public void postCommandProcessing( DataModelCommand command ) {
                             || ( (DataObjectChangeEvent) event ).getChangeType() == ChangeType.PACKAGE_NAME_CHANGE
                             || ( (DataObjectChangeEvent) event ).getChangeType() == ChangeType.CLASS_NAME_CHANGE ) {
 
-                        Annotation planningEntityAnnotation = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION );
-                        planningEntityAnnotation.setValue( "difficultyComparatorClass", dataObject.getName() + "." + dataObject.getName() + "Comparator.class" );
-
-                        plannerDataObjectEditorService.call( new RemoteCallback<ComparatorObject>() {
-                            @Override
-                            public void callback( ComparatorObject updatedComparatorObject ) {
-                                dataObject.getNestedClasses().clear();
-                                if ( updatedComparatorObject != null ) {
-                                    dataObject.addNestedClass( updatedComparatorObject );
-
-                                    comparatorDefinition.setValue( "fieldPaths", getAnnotationDef( updatedComparatorObject.getObjectPropertyPathList() ) );
-                                } else {
-                                    dataObject.removeAnnotation( comparatorDefinition.getClassName() );
-                                    Annotation planningEntityAnnotation = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION );
-                                    if ( planningEntityAnnotation != null ) {
-                                        planningEntityAnnotation.removeValue( "difficultyComparatorClass" );
+                        if ( ( (DataObjectChangeEvent) event ).getChangeType() == ChangeType.CLASS_NAME_CHANGE ) {
+                            dataModelerService.call( new RemoteCallback<DataObject>() {
+                                @Override
+                                public void callback( DataObject dataObject ) {
+                                    if ( dataObject != null ) {
+                                        comparatorObjectHandler.updateComparatorObject( dataObject, comparatorObject );
+
+                                        dataObject.getNestedClasses().clear();
+                                        dataObject.addNestedClass( comparatorObject );
+
+                                        Annotation planningEntityAnnotation = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION );
+                                        planningEntityAnnotation.setValue( "difficultyComparatorClass", dataObject.getName() + "." + dataObject.getName() + "Comparator.class" );
+
+                                        dataModelerService.call().saveSource( event.getSource(), event.getPath(), dataObject, null, "Update comparator object after data object class name change" );
```

Comment:
I'm afraid that we can't save the java file from this point independently from users decision. If we do this we can leave the file inconsistent. e.g. I can imagine a scenario  when the user opens the file, plays a bit (changing eventually the class name, etc.) and then closes the file by cancelling the modifications in the UI. If the file was automatically saved it will be inconsistent with the user desire.